### PR TITLE
Support x509

### LINKED ISSUE #79: Support for X.509 authentication
Floodgate should be able to authenticate using X.509 authentication as it is one of standard authentication mechanism used in Spinnaker.

### PR DESCRIPTION
**Pull request by kacpersaw**
Closes #79

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -36,11 +37,34 @@ type Config struct {
...
+			CertPath string `json:"certPath"`
+			KeyPath  string `json:"keyPath"`
+		} `json:"x509"`
 	} `json:"auth"`
 	Libraries []string `json:"libraries"`
 	Resources []string `json:"resources"`
 }
 
+// IsAuthValid is used to check if is only one auth method selected
+func (config *Config) IsAuthValid() bool {
```
Comment (mwierzbicki) [inline config/config.go:52]: What if no method is set in config? This will return true.

Comment (mwierzbicki) [reply config/config.go:52]: Ps. unit tests would catch that ;)

File: examples/config.yaml [hunk 2]
```diff
@@ -13,6 +13,10 @@ auth:
     clientSecret: xxxxxxxxxxxxxxxxxxxxxxxxxxxx
     scopes:
       - profile
+  x509:
+    certPath: /etc/ssl/certs/floodgate.crt
+    enabled: true
+    keyPath: /etc/ssl/private/floodgate.key
```
Comment (boscard) [inline examples/config.yaml:19]: Will relative path work here? If so, maybe better path in example file is something like `~/.config/floodgate/floodgate.key` ?

File: config/config.go [hunk 3]
```diff
@@ -1,8 +1,9 @@
 package config
```
[the remark this answers is not in the document]
Comment (kacpersaw) [reply config/config.go:1]: Look at `config/auth` package. I added unit test for IsValid function.

File: examples/config.yaml [hunk 4]
```diff
@@ -13,6 +13,10 @@ auth:
     clientSecret: xxxxxxxxxxxxxxxxxxxxxxxxxxxx
     scopes:
       - profile
+  x509:
+    certPath: /etc/ssl/certs/floodgate.crt
```
[the remark this answers is not in the document]
Comment (kacpersaw) [reply examples/config.yaml]: This is only client certificate

**Pull request merged**